index.bs

```diff
@@ -4115,6 +4147,811 @@ opened</dfn> steps given |window|, |type| and |message|.
+
+1. Let |body| be request's [=request/body=].
+
+1. If |body| is a [=byte sequence=], let |body size| be the length of that
+   sequence. Otherwise, if |body| is a [=/body=] then let |body size| be that
+   body's <a spec=fetch>length</a>.
+
+1. Let |headers size| be the size in bytes of |request|'s [=request/headers list=] when
+   serialized as mandated by [[HTTP11]].
+
+   Note: For protocols which allow header compression, this is the compressed
+   size of the headers, as sent over the network.
+
+1. Let |headers| be an empty list.
+
+1. Let |cookies| be an empty list.
+
+1. For each (|name|, |value|) in |request|'s [=request/headers list=]:
+
+  1. Append the result of [=get a header=] with |name| and |value| to |headers|.
+
+  1. If |name| is a [=byte-case-insensitive=] match for "<code>Cookie</code>" then:
+
+    1. For each |cookie| in the user agent's cookie store that are included in
+       |request|:
+
+       Note: [[COOKIES]] defines some baseline requirements for which cookies in
+       the store can be included in a request, but user agents are free to
+       impose additional constraints.
+
+       1. Append the result of [=get a cookie=] given |cookie| to |cookies|.
+
+1. Let |timings| be [=get the fetch timings=] with |request|.
+
+1. Return a map matching the <code>network.RequestData</code> production, with
+   the <code>request</code> field set to |request id|, <code>url</code> field
+   set to |url|, the <code>method</code> field set to |method|, the
+   <code>headers</code> field set to |headers|, the |cookies| field set to
+   |cookies|, and the <code>headersSize</code> field set to |headers size|,
+   the <code>bodySize</code> field set to |body size|, and the
+   <code>timings</code> field set to |timings|.
+
+</div>
+
+#### The network.ResponseContent type #### {#type-network-ResponseContent}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl">
+network.ResponseContent = {
+    size: js-uint
+};
+</pre>
+
+The <code>network.ResponseContent</code> type represents the decoded response to
+a network request.
+
+<!-- Not sure this is worthwhile if it only ends up with a single field, but it
+would be natural to add a field here if we have a way to return the body -->
+
+<div algorithm>
+To <dfn>get the response content info</dfn> given |response|.
+
+1. Return a new map matching the <code>network.ResponseContent</code>
+   production, with the <code>size</code> field set to |response|'s [=response body
+   info=]'s [=decoded size=]
+
+</div>
+
+#### The network.ResponseData type #### {#type-network-ResponseData}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl">
+network.ResponseData = {
+    url: text,
+    protocol: text,
+    status: js-uint,
+    statusText: text,
+    fromCache: bool,
+    headers: [*network.Header],
+    mimeType: text,
+    bytesReceived: js-uint,
+    headersSize: js-uint / null,
+    bodySize: js-uint / null,
```

Comment:
Note: It's reasonable to have the fields, but they are not implemented yet in CDP (the data length can be taken only with the data itself), so it would be nice to make the WPT tests not relate on this field. Same thing for the `ResponseData.content`.

Reply:
- `ResponseData.headersSize` and `ResponseData.bodySize` are optional, so we allow them to be null in the tests.
- `ResponseContent.size` is not supposed to be optional, but it seems we still allow it to be null in tests. 
- `RequestData.headersSize` is the only one expected by the tests to be non optional, so I imagine that's where you will have frequent issues

I feel like it's better to make progress here, so if it helps I would be in favor of having all those fields marked as optional for now so that we can land this PR, and then we can discuss each of them in separate issues. Then if we say that we keep a field in the spec, all vendors should try to support it

Reply:
> I feel like it's better to make progress here, so if it helps I would be in favor of having all those fields marked as optional for now so that we can land this PR, and then we can discuss each of them in separate issues. Then if we say that we keep a field in the spec, all vendors should try to support it

SGTM

Reply:
> If we want to challenge this further we need inputs from consumers for HAR files to know if they could do without the data.

My biggest concern is that those fields are not really needed or avoidable by users. If so, we are not going to implement them, which makes the spec not implementable. 

I hate blocking the progress of this PR, but that is what stopped me from approving it. After all, I guess the risk from our side is acceptable, so LGTM.